### ISSUE TITLE
async_pipe copy constructor broken on windows

### ISSUE THREAD

**Issue by randomguy3** [klemens-morgenstern/boost-process#157]
The move constructor might also be broken, and you can't assign the inherited handle to a `boost::asio::windows::stream_handle` in a child process.

The reason is, you can't assign a single handle (or any duplicates of that handle) to more than one `stream_handle` (or other asio handle) object. This is because it will add it to a [completion port](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365198(v=vs.85).aspx), and once you've assigned a handle to a completion port, that's it. You can't add it again, and you can't add it to another completion port. Doing so will produce a "The parameter is invalid" error.

There are two ways I see to deal with this: either lazily create the `stream_handle` objects as needed (which will cover most use cases - you can copy `async_pipe`s before writing to or reading from them), or doing something more clever when duplicating handles (don't call `DuplicateHandle`, but open the underlying pipe again).

**Comment by Invincibl-e** [comment]
@randomguy3 Did you found a solution to this problem? I saw that he finally called CreateIoCompletionPort but didn't quite understand why it failed.

**Comment by randomguy3** [comment]
I ended up creating a new class that creates a pipe in the same way as `async_pipe`, but doesn't assign the pipe ends to `stream_handle`s - it just holds the pipes directly (and closes them on destruction, duplicates them on copy, etc). I then also had to create an [extension](https://www.boost.org/doc/libs/1_68_0/doc/html/boost_process/extend.html) to assign it to stdin/stdout/stderr.

That worked well for my use case, but makes dealing with the local end of the pipe a bit more complicated.
